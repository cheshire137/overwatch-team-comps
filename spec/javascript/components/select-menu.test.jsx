import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import SelectMenu from '../../../app/assets/javascripts/components/select-menu.jsx'

describe('SelectMenu', () => {
  let component = null
  const item1 = { id: 1, name: 'First item' }
  let idSelected = item1.id
  const item2 = { id: 2, name: 'Second item' }
  const props = {
    items: [item1, item2],
    onChange: newID => { idSelected = newID },
    selectedItemID: idSelected,
    disabled: false,
    menuClass: () => 'custom-menu-class',
    containerClass: () => 'custom-container-class',
    menuToggleContents: () => <span>Menu</span>,
    menuItemClass: item => `item-${item.id}`,
    menuItemContent: (item, isSelected) => (
      <span className={isSelected ? 'with-selected' : ''}>
        <span className="css-truncate">{item.name}</span>
      </span>
    )
  }

  beforeEach(() => {
    component = <SelectMenu {...props} />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('can change selected item', () => {
    const rendered = shallow(component)

    const select = rendered.find('.menu-toggle')
    select.simulate('click', { target: { blur: () => {} } })

    const menuItemButton = rendered.find('.menu-item.item-2')
    menuItemButton.simulate('click', {
      preventDefault: () => {},
      target: { blur: () => {} }
    })

    expect(idSelected).toBe(item2.id)
  })
})
