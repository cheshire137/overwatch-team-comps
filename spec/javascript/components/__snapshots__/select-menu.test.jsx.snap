exports[`SelectMenu matches snapshot 1`] = `
<div
  className="menu-container custom-container-class">
  <button
    className="button menu-toggle"
    disabled={false}
    onClick={[Function]}
    type="button">
    <span>
      Menu
    </span>
     
    <i
      aria-hidden="true"
      className="fa fa-caret-down" />
  </button>
  <div
    className="menu custom-menu-class">
    <button
      className="menu-item button is-selected item-1"
      onClick={[Function]}>
      <span
        className="with-selected">
        <span
          className="css-truncate">
          First item
        </span>
      </span>
      <i
        aria-hidden="true"
        className="fa fa-check menu-item-selected-indicator" />
    </button>
    <button
      className="menu-item button item-2"
      onClick={[Function]}>
      <span
        className="">
        <span
          className="css-truncate">
          Second item
        </span>
      </span>
      
    </button>
  </div>
</div>
`;
